### PR TITLE
Document some construtors in transitions.dart

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 1]
```diff
@@ -320,6 +344,9 @@ typedef Widget TransitionBuilder(BuildContext context, Widget child);
 /// For simple cases without additional state, consider using
 /// [AnimatedWidget].
 class AnimatedBuilder extends AnimatedWidget {
+  /// Creates an animated builder.
+  ///
+  /// The animation argument is required.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/transitions.dart]: `[animation]`

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 2]
```diff
@@ -20,6 +20,9 @@ export 'package:flutter/rendering.dart' show RelativeRect;
 /// For more complex case involving additional state, consider using
 /// [AnimatedBuilder].
 abstract class AnimatedWidget extends StatefulWidget {
+  /// Creates an animated widget.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/transitions.dart]: actually this doesn't. it's abstract. you can't call it.

**Pull request merged**